src/napari_builtins/_qt/features_table.py

```diff
@@ -430,90 +456,185 @@ def _get_selection_event_for_layer(layer):
+                if hasattr(layer, 'events') and hasattr(
+                    layer.events, 'features'
+                ):
+                    layer.events.features.connect(self._on_features_change)
+                selection_event = self._get_selection_event_for_layer(layer)
+                if selection_event is not None:
+                    selection_event.connect(self._update_table_selected_cells)
+
+            # Show widgets and update table
             self._on_features_change()
             self.toggle.setVisible(True)
             self.save.setVisible(True)
             self.table.setVisible(True)
+            self.join_toggle.setVisible(len(self._selected_layers) > 1)
+            self.info.setText(
+                f'Features of "{[layer.name for layer in self._selected_layers]}"'
+            )
         else:
+            # Hide widgets and show appropriate message
             self.toggle.setVisible(False)
+            self.join_toggle.setVisible(False)
             self.save.setVisible(False)
             self.table.setVisible(False)
 
-        if self._active_layer is None:
-            self.info.setText('No layer selected.')
-        elif not hasattr(self._active_layer, 'features'):
-            self.info.setText(
-                f'"{self._active_layer.name}" has no features table.'
-            )
-        else:
-            self.info.setText(f'Features of "{self._active_layer.name}"')
+            # Determine message based on original selection
+            if len(self.viewer.layers.selection) > 0:
+                self.info.setText('Selected layers do not have features.')
+            else:
+                self.info.setText('No layer selected.')
 
     def _on_features_change(self):
+        """Update the table with the features of the currently selected layers."""
         # TODO: optimize for smaller changes?
+        join_type = 'inner' if self.join_toggle.isChecked() else 'outer'
         self.table.model().sourceModel().replace_data(
-            self._active_layer.features
+            self._build_multilayer_features_table(join=join_type)
         )
         self.table.resizeColumnsToContents()
-        self._on_layer_selection_changed()
+        self._update_table_selected_cells()
+
+    def _build_multilayer_features_table(
+        self, join: str = 'outer'
+    ) -> pd.DataFrame:
+        """Builds a features table for multiple layers."""
+        df_list = []
+        for layer in self._selected_layers:
+            # All layers in self._selected_layers are guaranteed to have features
+            if layer.features is not None:
+                if 'layer' not in layer.features.columns:
+                    layer.features['layer'] = layer.name
+                    layer.features['layer'] = layer.features['layer'].astype(
+                        'category'
+                    )
+                df_list.append(layer.features)
+        df = pd.concat(df_list, ignore_index=True, join=join)
```

Comment:
Retaining the indeces may be worthwhile (`ignore_index=True`), but it would mean that quering the table by index could yield multiple values returned. Values in the concatenated table would have to be queried by index *and* layer name.